### DISCUSSION TITLE
Looking for a C.groovy solution

### DISCUSSION THREAD

**Discussion by rsjrny** [IBM/dbb-zappbuild#366] [General]
I currently use IBM C and Dignus C in my products and need to be able to build these along with the rest of my Assembler and Cobol code.

Has anyone created a C solution?
I did take the cobol.groovy and modified it to execute Dignus C but it is not ready for primetime yet.

What solutions do you have?

**Comment by FALLAI-Denis** [comment]
Hi @rsjrny 

We have the same need, also for Z Open Editor and include (header) resolution. 

Not on our short roadmap to date. 
Creating a C.groovy build script should be close to the Cobol.groovy script. 
Does C compiler have coprocessor for DB2 and CICS ?
Our (very old) current cataloged procedures include a preprocessor for Db2 and CICS and also a prelink step : no more need since PDS-E wich is mandatory for COBOL V5/V6. 

I see IBM has delivered a new C compiler as a replacement candidate for IBM XL C/C++... 
More performances, more functionalities, more compatibility...
I wonder if this new compiler should be use for use in DBB (a change in a change...). 
To see if that new compiler supports "ddname list" wich is often a requirement to implement in DBB.

**Reply by Crosswind** [reply]
Are you referring to the Open XL C/C++ compiler being the "new C compiler"? My understanding is that the compiler is not intended to build legacy applications but much rather for porting software to USS. This is also mentioned in the docs somewhere. So the solution should target the existing C/C++ compiler. But that's details ;-)
